libs/collectives/tests/unit/CMakeLists.txt

```diff
@@ -4,22 +4,25 @@
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
-set(tests
-    all_gather
-    all_reduce
-    all_to_all
-    barrier
-    broadcast
-    broadcast_direct
-    broadcast_apply
-    broadcast_component
-    fold
-    gather
-    global_spmd_block
-    reduce
-    remote_latch
-    scatter
-)
+set(tests barrier fold)
```

Comment:
@hkaiser should `barrier` and `fold` perhaps also run on two localities?

Reply:
Yes, that might be a good idea.

Reply:
However all of those should also run when networking is off as the facilities should be functional on one locality as well.

Reply:
Yeah, fair point. I changed it this way because those tests assert that they run with at least two localities. We can remove those asserts instead.

Reply:
I removed the checks for at least two localities, let's see how they run.

Reply:
Thanks!

Reply:
@hkaiser it looks like broadcast and scatter time out when running only on one locality. Is that something you would expect to happen? Only `broadcast_to` gets called with one locality. Does it (or the completion of the future) block on someone receiving the broadcast with `broadcast_from`?

Reply:
No, this is not what should happen. I'll have a look, thanks!

Reply:
> Only broadcast_to gets called with one locality. Does it (or the completion of the future) block on someone receiving the broadcast with broadcast_from?

That's exactly what's happening. Let's disable those tests for one locality for now. This is for another PR, I presume.

Reply:
Thanks for confirming. Yes, I think that's best left for another PR. I opened https://github.com/STEllAR-GROUP/hpx/issues/4778 to keep track of that issue.

Reply:
I excluded scatter and broadcast again if networking isn't enabled. Let's see what CI says this time...

Reply:
@msimberg: thanks for creating the ticket.